src/edgeProxy/public/javascripts/uio.js

```diff
@@ -1,19 +1,166 @@
+fluid.defaults("fluid.prefs.edgeProxyStore", {
+    gradeNames: ["fluid.dataSource.noEncoding", "fluid.modelComponent"],
+    members: {
+        isFreshLogon: "@expand:fluid.prefs.edgeProxyStore.getFreshLogon()"
+    },
+    model: {
+        isLoggedIn: false
+    },
+    components: {
+        unauthedStore: {
+            type: "fluid.prefs.cookieStore",
+            options: {
+                writable: true
             }
         }
+    },
+    dynamicComponents: {
+        authedStore: {
+            source: "{that}.model.isLoggedIn",
+            type: "fluid.prefs.pdsStore"
+        }
+    },
+    listeners: {
+        "onCreate.updateLoggedInState": {
+            listener: "fluid.prefs.edgeProxyStore.updateLoggedInState",
+            args: ["{that}"]
+        },
+        "onRead.impl": {
+            listener: "fluid.prefs.edgeProxyStore.get",
+            args: ["{that}"]
+        },
+        "onWrite.impl": {
+            listener: "fluid.prefs.edgeProxyStore.set",
+            args: ["{that}", "{arguments}.0"]   // settings
+        }
+    },
+    modelListeners: {
+        isLoggedIn: {
+            listener: "fluid.prefs.edgeProxyStore.updateUnauthedSettings",
+            args: ["{that}", "{prefsEditorLoader}", "{change}.value"],
+            excludeSource: "init"
+        }
     }
 });
+
+// Update the fresh logon state at the page load. Check if PDS_freshLogon is a query parameter of the page url.
+// If yes, return true. Otherwise, return false.
+fluid.prefs.edgeProxyStore.getFreshLogon = function () {
+    const url = new URL(window.location);
+    const isFreshLogon = url.searchParams.get("PDS_freshLogon") === "true";
+
+    if (isFreshLogon) {
+        // Remove "PDS_freshLogon" from query strings not to confuse users
+        url.searchParams.delete("PDS_freshLogon");
+        window.history.replaceState({}, "", url.href);
+    }
+
+    return isFreshLogon;
+};
+
+// Update the log in states at the page load.
+fluid.prefs.edgeProxyStore.updateLoggedInState = function (that) {
+    that.applier.change("isLoggedIn", getCookieValue("PDS_loginToken") ? true : false);
+};
+
+fluid.prefs.edgeProxyStore.get = async function (that) {
+    if (that.isFreshLogon) {
+        const unauthedPrefs = await fluid.prefs.edgeProxyStore.getPrefsFromStore(that.unauthedStore);
+        const authedPrefs = await fluid.prefs.edgeProxyStore.getPrefsFromStore(that.authedStore);
+        return {
+            preferences: fluid.extend(true, {}, authedPrefs, unauthedPrefs)
+        };
+    } else {
+        return that[that.model.isLoggedIn ? "authedStore" : "unauthedStore"].get();
+    }
+};
+
+fluid.prefs.edgeProxyStore.set = function (that, settings) {
+    return that[that.model.isLoggedIn ? "authedStore" : "unauthedStore"].set({}, settings);
+};
+
+fluid.prefs.edgeProxyStore.getPrefsFromStore = async function (store) {
+    const settings = await store.get();
+    return settings && settings.preferences ? settings.preferences : {};
```

Comment:
If you want to try something new you could write this like:
```suggestion
    return settings?.preferences ?? {};
```

Reply:
> return settings?.preferences ?? {};

I tried the use of `??` before. It didn't pass linting. I didn't spend enough time looking into how to fix it.

Reply:
You'd need to update the parser in eslintrc.json to `"ecmaVersion": 2020` or later I believe.